homeassistant/components/opnsense/coordinator.py

```diff
@@ -0,0 +1,139 @@
+"""DataUpdateCoordinator for the OPNSense integration."""
+from __future__ import annotations
+
+from dataclasses import dataclass
+from datetime import timedelta
+import logging
+
+from pyopnsense import diagnostics
+from pyopnsense.exceptions import APIException
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    CONF_API_KEY,
+    CONF_NAME,
+    CONF_SCAN_INTERVAL,
+    CONF_TIMEOUT,
+    CONF_URL,
+    CONF_VERIFY_SSL,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+
+from .const import (
+    CONF_API_SECRET,
+    CONF_TRACKER_INTERFACE,
+    DEFAULT_SCAN_INTERVAL,
+    DEFAULT_TIMEOUT,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@dataclass(slots=True, frozen=True)
+class OPNSenseResult:
+    """Dataclass returned by the coordinator."""
+
+    hostname: str
+    ip_address: str
+    mac_address: str
+    manufacturer: str
+
+
+class OPNSenseUpdateCoordinator(DataUpdateCoordinator[dict[str, OPNSenseResult]]):
+    """The OPNSense update coordinator."""
+
+    interfaces_client: diagnostics.InterfaceClient
+    netinsight_client: diagnostics.NetworkInsightClient
+
+    hass: HomeAssistant
+    entry: ConfigEntry
+
+    name: str
+    url: str
+    api_key: str
+    api_secret: str
+    verify_ssl: bool
+    tracker_interfaces: list
+
+    def __init__(self, hass: HomeAssistant, entry: ConfigEntry) -> None:
```

Comment:
```suggestion
    def __init__(self, hass: HomeAssistant) -> None:
```
config_entry is already loaded in the coordinator in the `super().__init__()`